pkg/console/controllers/managedclusters/controller_test.go

```diff
@@ -0,0 +1,122 @@
+	cluster3.Status = v1.ManagedClusterStatus{
+		ClusterClaims: []v1.ManagedClusterClaim{
+			{Name: "product.open-cluster-management.io", Value: "Openshift"},
+			{Name: "version.openshift.io", Value: "3.9"},
+		},
+	}
+	clusters = append(clusters, cluster3)
+
+	cluster4 := v1.ManagedCluster{}
+	cluster4.ObjectMeta = metav1.ObjectMeta{Name: "ProductEKSVer311"}
+	cluster4.Status = v1.ManagedClusterStatus{
+		ClusterClaims: []v1.ManagedClusterClaim{
+			{Name: "product.open-cluster-management.io", Value: "EKS"},
+			{Name: "version.openshift.io", Value: "3.11.rcbeata"},
+		},
+	}
+	clusters = append(clusters, cluster4)
+
+	return clusters
+}
+
+func TestGenerateManagedClusterController(t *testing.T) {
+	clusters := generateClusters(t)
+	t.Logf("\n>>>>>>> Validating %d clusters have been created\n...", len(clusters))
+}
+
+func TestProductCheckForClusterList(t *testing.T) {
+	expectedCluster := []string{}
+	expectedCluster = append(expectedCluster, "ProductOpenshiftVer49")
+	expectedCluster = append(expectedCluster, "ProductOpenshiftVer39")
+	clusters := generateClusters(t)
+	validClusters := []string{}
+	if len(clusters) < 1 {
+		t.Fatalf("\n---------------\nNo Valid clusters, count was %v\n-----------", len(clusters))
+	}
+	for _, managedCluster := range clusters {
+		clusterName := managedCluster.GetName()
+		claims := util.ClaimsMap(&managedCluster)
+
+		// Omit clusters that have the following product name
+		omittedProductCluster := []string{"AKS", "EKS", "IKS", "ROKS"}
+		if ProductCheckForClusterList(claims, omittedProductCluster) {
+			t.Logf(">>>>> Skipping managed cluster %q, product %q is unsupported", clusterName, claims["product.open-cluster-management.io"])
+			continue
+		}
+		validClusters = append(validClusters, clusterName)
+	}
+
+	if diff := deep.Equal(validClusters, expectedCluster); diff != nil {
+		t.Error(diff)
+		t.Fatalf("\n---------------\nValid clusters are %v, while expected clusters are %v\n-----------", validClusters, expectedCluster)
+	}
+	t.Logf(">>>> Numer of Valid Clusters %v------- EOT\n", validClusters)
+
+}
+
+func TestVersionCheckForClusterList(t *testing.T) {
+	expectedCluster := []string{}
+	expectedCluster = append(expectedCluster, "ProductAKSVer412")
+	expectedCluster = append(expectedCluster, "ProductOpenshiftVer49")
+	clusters := generateClusters(t)
+	validClusters := []string{}
+	if len(clusters) < 1 {
+		t.Fatalf("\n---------------\nNo Valid clusters, count was %v\n-----------", len(clusters))
+	}
+	for _, managedCluster := range clusters {
+		clusterName := managedCluster.GetName()
+		claims := util.ClaimsMap(&managedCluster)
+
+		if VersionCheckForClusterList(claims) {
+			t.Logf("Skipping managed cluster %q with version %q, versions prior to openshift 4.0 not supported", clusterName, claims["version.openshift.io"])
+			continue
+		}
+		validClusters = append(validClusters, clusterName)
+	}
+	if diff := deep.Equal(validClusters, expectedCluster); diff != nil {
+		t.Error(diff)
```

Comment:
When this function is called the program will fail, so no other calls will get executed.